pumpkin/src/main.rs

```diff
@@ -111,105 +96,61 @@ fn main() -> io::Result<()> {
             });
         }
         loop {
```

Comment:
Although this code should work, I have no idea why it doesn't. The packets are being read but aren't being handled for some reason.

Reply:
This code is bit older. I moved to a new design where i only actullay read the packets and process them in the main. See https://github.com/Snowiiii/Pumpkin/blob/master/pumpkin/src/main.rs#L157. When you only process in the poll client function no player packets will be handled

Reply:
I see, I'll take a look. Thank you!

Reply:
Also yep I'm aware that no player packets are handled, I'm just starting by trying to get status packets working.

Reply:
The code should work then

Reply:
Thats really weird, it's not working for some reason. I'll continue trying to find the problem.

Reply:
theres not a deadlock in there is there?

Reply:
Nope, there shouldn't be a deadlock. After further testing, I realized that the client is never sending a status request packet. Not sure what that means as the client should be sending it immediately after the handshake.

Reply:
Are packets being dropped by us between the creation of the client and the polling of the client?

Reply:
Its a possibility that packets are being lost on the polling of the client, I'll do some more testing.

Reply:
Alright I have tested and determined that the packet is not being handled on the server. Looking at Wireshark the client is definitely sending the packet to the server.  The server always handles the handshake (switches states). The server correctly deserializes the send packet. But for some reason when handling the packet, the server never responds to the status packet.

Reply:
Had this myself aswell

Reply:
Alright, I think the next step is to use a debugger to see what's going on.

Reply:
@neeleshpoli @kralverde 
The recent PR i made solves the locking issue. This was also the reason me and lukas stopped the port to tokio. I already did much debugging when porting to tokio. The code i gave you should work mostly and it did work. I had a Status response and could Join the World.

Reply:
Alright, thank you. I'll finish porting to tokio